### ISSUE TITLE
Restart code requires input and output file formats to be h5

### ISSUE THREAD

**Issue by mattzett** [gemini3d/gemini3d#31]
Even if the milestones are written as .h5 the find_milestone() subroutine uses the input file format to determine whether restarting is possible.  So if you happen to use binary input and h5 output (restarting should be possible in principle) it will refuse to restart due to binary file inputs...  Generally, we should probably all just migrate to h5 for everything so we don't have to keep fighting with formats, etc...

**Comment by scivision** [comment]
It should be easy to make milestones work with NetCDF4 as well. Gemini allows one to load HDF5 and save NetCDF or vice versa. The main practical limitation I've seen is that some Linux distributions e.g. Ubuntu 18.04 have a problem with their libhdf5-dev + libnetcdff-dev where linking both into the same program causes bizarre errors. This is when I updated `gemini3d.compile_prereqs` to build HDF5 and NetCDF4 libraries themselves in order, and I found they worked. <-- I had been worried it was a problem in h5fortran + nc4fortran but it was actually a Linux distro library problem.

In short, once someone wants this features it's easy to add NetCDF milestones.

**Comment by mattzett** [comment]
I think we've decided to deprecate non-hdf5 inputs so this is likely no longer an issue moving forward.
